paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -148,8 +146,55 @@ internal class CustomerSheetViewModel(
         scope = viewModelScope.plus(workContext)
     )
 
+    private val isEditing = MutableStateFlow(false)
+    private val selectionConfirmationState = MutableStateFlow(
+        SelectionConfirmationState(
+            isConfirming = false,
+            error = null,
+        )
+    )
+    private val customerState = MutableStateFlow(
+        CustomerState(
+            paymentMethods = listOf(),
+            configuration = configuration,
+            currentSelection = originalPaymentSelection,
+            metadata = null,
+        )
+    )
+
+    private val selectPaymentMethodState = combineAsStateFlow(
+        customerState,
+        selectionConfirmationState,
+        isEditing,
+    ) { customerState, selectionConfirmationState, editing ->
+        val paymentMethods = customerState.paymentMethods
+        val paymentMethodMetadata = customerState.metadata
+        val paymentSelection = customerState.currentSelection
+
+        val userCanEditAndIsEditing = editing && customerState.canEdit
+        val primaryButtonVisible = !userCanEditAndIsEditing && originalPaymentSelection != paymentSelection
+
+        CustomerSheetViewState.SelectPaymentMethod(
+            title = configuration.headerTextForSelectionScreen,
+            savedPaymentMethods = paymentMethods,
+            paymentSelection = paymentSelection,
+            canEdit = customerState.canEdit,
+            isLiveMode = isLiveModeProvider(),
+            canRemovePaymentMethods = customerState.canRemove,
+            isCbcEligible = customerState.cbcEligibility is CardBrandChoiceEligibility.Eligible,
+            primaryButtonVisible = primaryButtonVisible,
+            isGooglePayEnabled = paymentMethodMetadata?.isGooglePayReady == true,
+            isEditing = userCanEditAndIsEditing,
+            isProcessing = selectionConfirmationState.isConfirming,
+            errorMessage = selectionConfirmationState.error,
+            mandateText = paymentSelection?.mandateText(
+                merchantName = configuration.merchantDisplayName,
+                isSetupFlow = false,
+            )?.takeIf { primaryButtonVisible }
+        )
+    }
```

Comment:
I'm opting to maintain a select state rather than fragment the logic for it across multiple functions. When we move to interactors, we should no longer need to maintain a state as a flow.